### DISCUSSION TITLE
response.tryEnd doesn't copy the received buffer?

### DISCUSSION THREAD

**Discussion by uasan** [uNetworking/uWebSockets.js#983] [Q&A]
I noticed that tryEnd called on onWritable events does not copy the received buffer, i.e. If, after calling tryEnd, you write data to that part of the buffer that has already been sent, there is a chance that the client will receive changed data, because tryEnd did not copy the contents of the buffer.

**Comment by uasan** [comment]
Or the second assumption, the bufferOffset is not taken, i.e. you sometimes copy buffer from position zero

**Comment by uNetworkingAB** [comment]
You can see https://github.com/uNetworking/uWebSockets.js/blob/master/examples/VideoStreamer.js for an example on tryEnd

**Comment by uasan** [comment]
Yes, they use the `slice` method, it makes copies of bytes, I used method `subarray`, it does not copy bytes, I did not know that `tryEnd` does not copy the buffer, now I will know
